### PR TITLE
Gtk theme new changes from upstream

### LINKED ISSUE #2194: Button in GTK3 dialogue moves a little when hovered
#### Steps to reproduce the issue
1. Open gedit
2. Write text
3. Close gedit
4. Hover the mouse pointer on the Save button

**Probably the same issue and fix in Adwaita:**
https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/1948

gtk_dialogue_button

#### Software that presents the issue
GTK3 dialogues

#### Yaru version
$ apt show yaru-theme-gtk 
Package: yaru-theme-gtk
Version: 20.04.6

### PR DESCRIPTION
**Pull request by Feichtmeier**
- pulls branch upstreaam-gtk-changes, so the upstream folder is synced

Yaru src changes:
- fixes for linked buttons from upstream
- simplifications for linked buttons from upstream
- upstream introduced a new checkradio_borders_color and linked checkradio to switch color, so it looks like we do not need to overwrite the switch styling again, thus this section is removed in _tweaks
- needed to apply a tiny change to the switch styling, as it looks like a typo to me but they did not notice since they use the same blue for both selected and checkradio colors. Thus changing the backdrop border color for switches to checkradio_borders_color in _common, already reported upstream

grafik

grafik

grafik

@clobrano please test this, even though I already tested it:

- [x] gtk dialog  button roundings on hover/checked

- [x] checkradio colors in default/light/dark themes on active/backdrop

- [x] swicth colors in default/light/dark themes on active/backdrop and headerbars

Closes #2194 #2193

### CODE REVIEW

File: gtk/src/default/gtk-3.20/_common.scss [hunk 1]
```diff
@@ -168,7 +168,7 @@ flowbox {
   background-color: if($variant=='light', transparent, black);
   border-radius: 0;
   padding: 0;
-
+
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:171]: empty space can be removed

File: gtk/src/default/gtk-3.20/_common.scss [hunk 2]
```diff
@@ -1639,10 +1620,10 @@ headerbar {
       background-color: $suggested_bg_color;
       background-image: none;
       box-shadow: inset 0 1px mix($top_hilight, $suggested_bg_color, 60%);
-
-      label {
+
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:1623]: empty space can be removed

File: gtk/src/default/gtk-3.20/_common.scss [hunk 3]
```diff
@@ -1639,10 +1620,10 @@ headerbar {
       background-color: $suggested_bg_color;
       background-image: none;
       box-shadow: inset 0 1px mix($top_hilight, $suggested_bg_color, 60%);
-
-      label {
+      
+      label {
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:1624]: emtpy space

File: gtk/src/default/gtk-3.20/_common.scss [hunk 4]
```diff
@@ -1639,10 +1620,10 @@ headerbar {
       background-color: $suggested_bg_color;
       background-image: none;
       box-shadow: inset 0 1px mix($top_hilight, $suggested_bg_color, 60%);
-
-      label {
+      
+      label { 
         text-shadow: none;
-        color: $selected_fg_color;
+        color: $selected_fg_color;
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:1626]: empty space

File: gtk/src/default/gtk-3.20/_common.scss [hunk 5]
```diff
@@ -2881,12 +2863,12 @@ switch {
 
     @if $variant == 'light' {
       @include button(normal-alt, $edge: $shadow_color);
-    }
+    }
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:2866]: empty space

File: gtk/src/default/gtk-3.20/_common.scss [hunk 6]
```diff
@@ -2881,12 +2863,12 @@ switch {
...
     @if $variant == 'light' {
       @include button(normal-alt, $edge: $shadow_color);
-    }
+    } 
     @else {
       @include button(normal-alt, $c: lighten($bg_color,6%), $edge: $shadow_color);
     }
   }
-
+
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:2871]: empty space

File: gtk/src/default/gtk-3.20/_common.scss [hunk 7]
```diff
@@ -3033,11 +3015,11 @@ radio {
   }
 
   &:backdrop { transition: $backdrop_transition; }
-
+
```
Comment (clobrano) [inline gtk/src/default/gtk-3.20/_common.scss:3018]: empty space

**Comment by clobrano** [comment]
Looks good to me, just remove all the unnecessary whitespaces ;)

**Comment by Feichtmeier** [comment]
hm most whitespaces should come from upstream. isnt it better to have less diff? makes it easier for me next time there is a change

**Comment by clobrano** [comment]
As you prefer

**Comment by Feichtmeier** [comment]
I mean I totally agree that we should avoid those useless whitespaces, it's just that I copied the upstream files and then re-applied slim mode + border radii. 
What was your impression concerning the switches and checkradios? Any visible changes?

**Comment by clobrano** [comment]
> What was your impression concerning the switches and checkradios? Any visible changes?

No, I didn't notice any difference

**Pull request merged**